src/merkle/smt/full/mod.rs

```diff
@@ -294,3 +299,66 @@ impl From<&RpoDigest> for LeafIndex<SMT_DEPTH> {
         Word::from(value).into()
     }
 }
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for Smt {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        // serialize storage maps
+        let filled_leaves = self
+            .entries()
+            .filter(|(_, value)| *value != Smt::EMPTY_VALUE)
```

Comment:
I think this is not needed. The SMT should never have the `EMPTY` value in the entries, they are implied"

https://github.com/0xPolygonMiden/crypto/blob/b4f6d91e51cb716049ec7d14ed3f4a48bd64e849/src/merkle/smt/full/tests.rs#L311-L330

Reply:
you are right, I didn't think that through properly. Changed it.